fpsd/features.py

```diff
@@ -0,0 +1,765 @@
+        9          | 11        | 11        ...
+        10         | 12        | 14        ...
+        11         | 13        | 16        ...
+
+        where there are a variable number (num_features) of
+        feature columns, each named num_outgoing_packets_in_window_x_of_size_y.
+        num_outgoing_packets_in_window_x_of_size_y is the number
+        of cells in the xth bin or "window" of size y
+
+        Args:
+            num_features [int]: number of features to create
+            size_window [int]: size of each bin
+
+        Returns:
+            [string] name of newly created table
+        """
+
+        self._create_temp_packet_positions(outgoing_only=False)
+        feature_table_name = "features.size_{}_windows".format(size_window)
+
+        self.drop_stale_feature_table(feature_table_name)
+
+        if num_features > 1:
+            feature_columns = ["num_outgoing_packets_in_window_{}_of_size_{}".format(x,
+                                                                                     size_window)
+                               for x in range(1, num_features + 1)]
+
+            # Use LEFT OUTER JOIN because many of the later windows
+            # will be Null.
+            # Note: count(*) will return Null if count(*) = 0
+            arr_subqueries = [("LEFT OUTER JOIN (SELECT exampleid,          "
+                               "COALESCE(count(*), 0)                       "
+                               "AS {colname} FROM packet_positions WHERE    "
+                               "ingoing = false AND rank > {rank_start} AND "
+                               "rank <= {rank_stop} GROUP BY exampleid)     "
+                               "t{feat_ind} ON foo.exampleid =               "
+                               "t{feat_ind}.exampleid").format(colname=feature_columns[x-1],
+                                                               rank_start=(x-1)*size_window,
+                                                               rank_stop=x*size_window,
+                                                               feat_ind=x)
+                              for x in range(1, num_features + 1)]
+
+            subqueries = " ".join(arr_subqueries)
+        else:
+            subqueries = ""
+
+        query = ("CREATE TABLE features.size_{}_windows AS ( "
+                 "SELECT foo.exampleid, {} FROM ( "
+                 "(SELECT exampleid, count(*) FROM " 
+                 "raw.frontpage_traces GROUP BY exampleid) foo"
+                 " {} ));").format(size_window,
+                               ', '.join(feature_columns),
+                               subqueries)
+
+        self.execute_query_from_string(query)
+        return feature_table_name
+
+    def create_bursts(self):
+        """This method takes all examples and produces a
+        table public.current_bursts with all bursts in
+        the following format:
+
+        burstid   | burst     | exampleid   | rank
```

Comment:
This is intentional as in this case the primary key is `burstid` (for every `exampleid` there are many `burstid`s). By convention, primary key is usually the leftmost column